rsp.ld

```diff
@@ -0,0 +1,50 @@
+/* 
+ * rsp.ld: Linker script for rsp ucode. 
+ */
+
+OUTPUT_FORMAT ("elf32-bigmips", "elf32-bigmips", "elf32-littlemips")
+OUTPUT_ARCH (mips)
+ENTRY (_start)
+
+MEMORY
+{
+    /* This is the layout in ROM. */
+    rom_dmem : ORIGIN = 0x0000, LENGTH = 0x1000
+    rom_imem : ORIGIN = 0x1000, LENGTH = 0x1000
+
+    /* This is a workaround to make ld place text symbols at the correct addresses (0x0 - 0x1000).
+       The RSP technically uses a harvard-architecture (https://en.wikipedia.org/wiki/Harvard_architecture)
+       which means that it uses different address spaces for instructions and data accesses.
+       Because ld is not designed for such architectures, we need to place the data section somewhere different,
+       since it would otherwise overlap the text section. As a workaround, we place it at 0x04000000 (which is also
+       the location of DMEM from the VR4300's point of view). Because the RSP only uses the lower 12 bits
+       of any address, this works out fine (as long as we always wrap data addresses in "%lo()").
+       
+       Note that this is not actually required to run the ucode correctly (instruction addresses above 0x1000 are truncated anyway),
+       but it makes debugging with gdb a lot easier (e.g. using this fork of cen64 https://github.com/lambertjamesd/cen64).
```

Comment:
Yes I think so, but we would have to ask Snacchus about this actually.

Reply:
Yes this is actually somewhat specific to cen64. It will report DMEM/IMEM at `0x04000000`, but only to the debugger. This address is derived from the uncached virtual address `0xA4000000` where DMEM/IMEM is mirrored on hardware. But `0x04000000` doesn't actually make sense in any other context. We could change it to the aforementioned virtual uncached address, but then RSP-debugging with cen64 as-is will break slightly. This might not be a bad thing, because to my knowledge, I'm basically the only person who uses that specific feature, and I could just modify cen64.

Reply:
If this maps well to the actual memory map and not cen64 specific, let's keep it

Reply:
Ah sorry I haven't read it properly. Looks like the correct approach here is using the unchaced address. OTOH I am ok to have this as is right now and open respective issues here and cen64 and resolve later.

Reply:
Done